test/parallel/test-buffer-tostring-binary.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const assert = require('assert');
+
+// v8 fails and node process crashes if binary length > kMaxLength (268435440 bytes)
+const kMaxLength = (256 * 1024 * 1024) - 16;
+
+var maxString = Buffer(kMaxLength).toString('binary');
+assert.equal(maxString.length, kMaxLength);
+
+maxString = Buffer(kMaxLength + 1).toString('binary');
+assert.equal(maxString, undefined);
```

Comment:
Also make sure to take a look at `test/parallel/test-stringbytes-external-at-max.js` on master. There's some fail logic that's used in the rasp builds since they have so little memory.